odoo/models.py

```diff
@@ -1709,22 +1704,28 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu
         :raise Invalid ArchitectureError: if there is view type other than form, tree, calendar, search etc... defined on the structure
         """
         self.check_access_rights('read')
-        view = self.env['ir.ui.view'].sudo().browse(view_id)
 
         # Get the view arch and all other attributes describing the composition of the view
-        result = self._fields_view_get(view_id=view_id, view_type=view_type, toolbar=toolbar, submenu=submenu)
+        arch, view = self._get_view(view_id=view_id, view_type=view_type, **options)
 
         # Override context for postprocessing
-        if view_id and result.get('base_model', self._name) != self._name:
-            view = view.with_context(base_model_name=result['base_model'])
+        if view and (view.model or self._name) != self._name:
+            view = view.with_context(base_model_name=view.model)
 
         # Apply post processing, groups and modifiers etc...
-        xarch, xfields = view.postprocess_and_fields(etree.fromstring(result['arch']), model=self._name)
-        result['arch'] = xarch
-        result['fields'] = xfields
+        arch, models = view.postprocess_and_fields(arch, model=self._name, **{'mobile': options.get('mobile')})
```

Comment:
Why don't you pass all options?
In case that was your intention, please simplify the code
```suggestion
        arch, models = view.postprocess_and_fields(arch, model=self._name, mobile=options.get('mobile'))
```